packages/gatsby/src/schema/infer-graphql-input-fields.js

```diff
@@ -216,12 +216,14 @@ type InferInputOptions = {
 
 const recursiveOmitBy = (value, fn) => {
   if (_.isObject(value)) {
+    let newValue = {}
```

Comment:
`value` can be array (`_.isObject([])` is `true` - gotta love JS :D) - need to create same structure here